xoa/data_source_vms.go

```diff
@@ -0,0 +1,107 @@
+package xoa
+
+import (
+	"log"
+	"strings"
+
+	"github.com/ddelnano/terraform-provider-xenorchestra/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceXoaVms() *schema.Resource {
+
+	return &schema.Resource{
+		Read: dataSourceVmsRead,
+		Schema: map[string]*schema.Schema{
+			"vms": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     resourceVm(),
+			},
+			"num": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
```

Comment:
See my comment below about this. I'm not sure this provides much value when the `length` function can be leveraged.